dev/ProgressRing/ProgressRing.cpp

```diff
@@ -49,38 +34,15 @@ void ProgressRing::OnApplyTemplate()
 {
     winrt::IControlProtected controlProtected{ *this };
 
-    m_outlinePath.set(GetTemplateChildT<winrt::Path>(s_OutlinePathName, controlProtected));
-    m_outlineFigure.set(GetTemplateChildT<winrt::PathFigure>(s_OutlineFigureName, controlProtected));
-    m_outlineArc.set(GetTemplateChildT<winrt::ArcSegment>(s_OutlineArcName, controlProtected));
-    m_ringPath.set(GetTemplateChildT<winrt::Path>(s_RingPathName, controlProtected));
-    m_ringFigure.set(GetTemplateChildT<winrt::PathFigure>(s_RingFigureName, controlProtected));
-    m_ringArc.set(GetTemplateChildT<winrt::ArcSegment>(s_RingArcName, controlProtected));
-
     m_player.set([this, controlProtected]()
     {
-        auto const player = GetTemplateChildT<winrt::AnimatedVisualPlayer>(s_LottiePlayerName, controlProtected);
-        if (player)
-        {
-            player.RegisterPropertyChangedCallback(winrt::UIElement::OpacityProperty(), { this, &ProgressRing::OnOpacityPropertyChanged });
-        }
+        auto const player = GetTemplateChildT<winrt::AnimatedVisualPlayer>(s_IndeterminateAnimatedVisualPlayerName, controlProtected);
```

Comment:
>player [](start = 19, length = 6)

delete the lambda here, it doesn't do anything

Reply:
actually, SetANimatedVisualPlayerSource is a helper that is only ever called here, and it immediately calls m_player.get so I think it would be better to copy the body of that helper into this lambda and then delete the helper function 

---
In reply to: [401145059](https://github.com/microsoft/microsoft-ui-xaml/pull/2197#discussion_r401145059) [](ancestors = 401145059)

Reply:
I think it's better to keep it in the helper function for now. As Determinate and other Lottie theme properties are added in in subsequent updates, it'll become a much larger helper function?